waiter/src/waiter/cors.clj

```diff
@@ -147,3 +148,49 @@
   "Creates a CORS validator that denies all cross-origin requests."
   [_]
   (->DenyAllCorsValidator))
+
+(defrecord TokenParametersCorsValidator [allow-cors?]
+  CorsValidator
+  (preflight-allowed? [_ request] (allow-cors? request))
+  (request-allowed? [_ request] (allow-cors? request)))
+
+(defn- allowed-cors-rule-matches?
+  "Checks if an allowed CORS rule matches"
+  [origin-scheme origin-no-host target-scheme path method
+   [_ {:strs [origin-regex origin-schemes target-path-regex target-schemes methods] :as xxx}]]
+  (cond
+    (and methods (not (.contains methods (str/upper-case (name method))))) false
```

Comment:
`(not (.contains methods (str/upper-case (name method))))` =>
`(not-any? #(= % (str/upper-case (name method))) methods)`

Reply:
Same for the expressions below